File: P2-Thrift/cliente.py
from calculadora import Calculadora

from thrift import Thrift
from thrift.transport import TSocket
from thrift.transport import TTransport
from thrift.protocol import TBinaryProtocol

transport = TSocket.TSocket("localhost", 9090)
transport = TTransport.TBufferedTransport(transport)
protocol = TBinaryProtocol.TBinaryProtocol(transport)

client = Calculadora.Client(protocol)

transport.open()

print("hacemos ping al server")
client.ping()

resultado = client.suma(2.5, 2.5)
print("2.5 + 2.5 = " + str(resultado))
resultado = client.resta(2.5, 2.5)
print("2.5 - 2.5 = " + str(resultado))
resultado = client.multiplica(2.5, 2.5)
print("2.5 * 2.5 = " + str(resultado))

try:
    resultado = client.divide(2.5, 0)
    print("2.5 / 2.5 = " + str(resultado))
except Calculadora.OperacionInvalida as e:
    print(e.mensaje)

resultado = client.divide(2.5, 2.5)
print("2.5 / 2.5 = " + str(resultado))

transport.close()

File: P2-Thrift/cliente.rb
require 'thrift'

require_relative 'calculadora'
 
begin 
    transport = Thrift::BufferedTransport.new(Thrift::Socket.new('localhost', 9090))
    protocol = Thrift::BinaryProtocol.new(transport)
    client = Calculadora::Client.new(protocol)
 
    transport.open()
 
    client.ping()

    result = client.suma(2.5,2.5)

    puts result

    transport.close()
 
rescue Thrift::Exception => tx
    print 'Thrift::Exception: ', tx.message, "\n"
end